Update YottaDB install version

josephsnyder opened the pull request:
With a recent update to the Yotta installer, 1.10 was no longer
available.  Update to a more recent version and ensure that a necessary
program is added.

To eliminate issues with a constantly updating file, remove the SHA1
check.

josephsnyder committed:
Update YottaDB install version

With a recent update to the Yotta installer, 1.10 was no longer
available.  Update to a more recent version and ensure that a necessary
program is added.

To eliminate issues with a constantly updating file, remove the SHA1
check.

ChristopherEdwards commented on GTM/install.sh:
```diff
@@ -86,13 +86,6 @@ fi
 curl -s -L https://raw.githubusercontent.com/YottaDB/YottaDB/master/sr_unix/ydbinstall.sh -o ydbinstall
 
-# Verify hash as we are going to make it executable
-sha1sum -c --status ydbinstall_SHA1
```
I'm torn about removing this - yes it removes an issue when ydbinstall is updated, but if the download fails for whatever reason and downloads an http file and executes it - it won't work. maybe we should grab the first line of the file and make sure it matches a regex?

shabiel commented:
It won't work when it's updated; and it won't work if it is not found.
Which is the lesser evil?

--Sam

On Wed, Sep 5, 2018 at 3:01 PM Christopher Edwards <notifications@github.com>
wrote:

> *@ChristopherEdwards* commented on this pull request.
> ------------------------------
>
> In GTM/install.sh
> <https://github.com/OSEHRA/docker-vista/pull/14#discussion_r215389217>:
>
> > @@ -86,13 +86,6 @@ fi
>  echo "Downloading ydbinstall"
>  curl -s -L https://raw.githubusercontent.com/YottaDB/YottaDB/master/sr_unix/ydbinstall.sh -o ydbinstall
>
> -# Verify hash as we are going to make it executable
> -sha1sum -c --status ydbinstall_SHA1
>
> I'm torn about removing this - yes it removes an issue when ydbinstall is
> updated, but if the download fails for whatever reason and downloads an
> http file and executes it - it won't work. maybe we should grab the first
> line of the file and make sure it matches a regex?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/OSEHRA/docker-vista/pull/14#pullrequestreview-152642733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABEtwkvYRMDgsWSih7JIkNX4PoUr9q6Aks5uYB9wgaJpZM4WbSbo>
> .
>

ChristopherEdwards commented:
Right, but having it fail in a sane way is much preferred than a crash and burn from bash trying to execute HTML...

shabiel commented:
It fails in a sane way right now??? It just continues installing VistA like
nothing is wrong...

On Wed, Sep 5, 2018 at 3:09 PM Christopher Edwards <notifications@github.com>
wrote:

> Right, but having it fail in a sane way is much preferred than a crash and
> burn from bash trying to execute HTML...
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/OSEHRA/docker-vista/pull/14#issuecomment-418846966>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABEtwnmYOkSuKSkq8i3CIDc077EQfdf0ks5uYCFYgaJpZM4WbSbo>
> .
>

ChristopherEdwards commented:
Well technically that is another bug :)

ChristopherEdwards commented:
it does indeed stop this script as it does check the status code, the outer script needs to check for the failure too.

ChristopherEdwards commented:
if it is that much of an issue i'll just approve and fix later/file an issue.

josephsnyder commented:
Readding Sam.  Sorry about the email spam, Christopher.

> I don't think it is that big of a deal.  I generated this fix as a 
> first pass for the Vagrant install before we decided to remove the 
> check completely
>
>> curl -s -L 
>> https://raw.githubusercontent.com/YottaDB/YottaDB/master/sr_unix/ydbinstall.sh 
>> -o ydbinstall
>> isValidFile=`head ydbinstall | grep "Fidelity National Information"`
>> if [[ ! $isValidFile ]]; then
>
> Would that be enough of a check?
>
> - Joe

Joseph Snyder
R&D Engineer

Kitware, Inc.
28 Corporate Dr
Clifton Park, NY 12065
(518) 881-4409

On 9/5/2018 3:14 PM, Christopher Edwards wrote:
>
> if it is that much of an issue i'll just approve and fix later/file an 
> issue.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/OSEHRA/docker-vista/pull/14#issuecomment-418848682>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAhFmojhu8LMQLh77pzjNM2bsng_66GOks5uYCKPgaJpZM4WbSbo>.
>

josephsnyder committed:
Introduce different check for YDB file

Introduce a check which ensures that the file is the install and not the
HTML 404 page.

shabiel commented:
@josephsnyder: You can make that fix and I can test again.

On Wed, Sep 5, 2018 at 3:22 PM Christopher Edwards <notifications@github.com>
wrote:

> @josephsnyder <https://github.com/josephsnyder> That is the fix i had in
> mind, we can delay making sure vista doesn't install if this fails in the
> parent script as a separate issue.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/OSEHRA/docker-vista/pull/14#issuecomment-418851182>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABEtwotouuG0A402mngVHyffLWQXCdExks5uYCR8gaJpZM4WbSbo>
> .
>
